src/cargo/util/context/mod.rs

```diff
@@ -523,6 +526,16 @@ impl GlobalContext {
             .borrow_mut()
     }
 
+    /// Local overlay configuration.
+    ///
+    /// The values in this map are paths pointing to a local registry. Inserted overlays will
+    /// affect any newly created `PackageRegistry`s.
+    ///
+    /// See [`crate::sources::overlay::OverlayConfusionAttack`] for why you shouldn't use this.
+    pub fn local_overlays(&self) -> RefMut<'_, HashMap<SourceId, PathBuf>> {
```

Comment:
I'd hope we can avoid putting this on GlobalContext but its hard to say without reviewing the follow up at the same time.  Should we defer this to the follow up PR?

My motivation is that it gets messy adding and removing from the Global Context and if we can directly instantiate things, the easier it will be to maintain.

Reply:
I can give a little more context before we decide whether or not to defer. `cargo package` needs the overlay to take effect during a call to `ops::compile`. AFAICT there are three possibilities:

- put the overlay configuration in `GlobalContext`
- put the overlay configuration in the ephemeral workspace that `cargo package` creates
- have `cargo package` do something lower-level than `ops::compile`

The workspace option has the advantage that the overlay configuration doesn't need to be removed (because the workspace is discarded). It seems like it would just require a little work in `resolve_ws` and `resolve_ws_with_opts` to get the overlay configuration out of the workspace and insert it into the `PackageRegistry`

Reply:
The workspace seems the most beneficial.  Whichever way we go, still seems like it could be split out into a follow up PR